app/src/main/java/org/mozilla/fenix/components/Push.kt

```diff
@@ -40,7 +43,22 @@ class Push(context: Context, crashReporter: CrashReporter) {
 
         logger.debug("Creating push configuration for autopush.")
         val projectId = context.resources.getString(resId)
-        PushConfig(projectId)
+        val serverOverride = context.settings().overridePushServer
```

Comment:
I'm surprised this line doesn't invoke strict mode. Were you able to test this patch with a first-run install and a synced account setup?